File: 04_make_mks/euler.c
#include "euler.h"
#include <math.h>

int multiples_15(int limit) {
  int x, result = 0;
  for (x = 2; x < limit; ++x) {
    if (x % 5 == 0 || x % 3 == 0) {
      result += x;
    }
  }
  return result;
}

int smallest_multiple(int limit) {
  unsigned int num = 2520, fix = 2520;
  int current_num = 11;
  while (current_num < limit) {
    if (num % current_num == 0) {
      fix = num;
      current_num++;
    } else {
      num += fix;
    }
  }
  return num;
}

int sum_square_difference(int limit) {
  long result = 0;
  int num;
  for (num = 1; num < limit; ++num) {
    result += pow(num, 3) - pow(num, 2);
  }
  return result;
}

File: 04_make_mks/rust/euler.rs
// I actually don't this moduler approach for rust
pub fn multiples_15() -> u32 {
    let mut result = 0;
    for x in 2..1000 {
        if x % 5 == 0 || x % 3 == 0 {
           result += x; 
        }
    }
    return result;
}

pub fn smallest_multiple() -> u32 {
    let mut current_num: u8 = 11;
    let mut num: u32 = 2520;
    let mut fix: u32 = 2520;
    while current_num < 21 {
        if num % current_num as u32 == 0 {
            fix = num;
            current_num += 1;
        } else {
            num += fix;
        }
    }
    return num;
}

pub fn sum_square_difference() -> i32 {
    let mut result: i32 = 0;
    for num in 1i32..101 {
       result += num.pow(3) - num.pow(2); 
    }
    return result;
}
